src/prevector.h

```diff
@@ -166,10 +167,15 @@ class prevector {
             }
         } else {
             if (!is_direct()) {
+                /* FIXME: Because malloc/realloc here won't call new_handler if allocation fails, assert
+                    success. These should instead use an allocator or new/delete so that handlers
+                    are called as necessary, but performance would be slightly degraded by doing so. */
                 _union.indirect = static_cast<char*>(realloc(_union.indirect, ((size_t)sizeof(T)) * new_capacity));
+                assert(_union.indirect);
```

Comment:
Would prefer:
```
if (!_union.indirect) { new_handler_terminate(); }
```
(and change the comment appropriately).

Reply:
... except that ``new_handler_terminate`` is static in ``init.cpp``, but that can easily be fixed.

Reply:
I'm not particularly keen on this, because currently `prevector.h` has no dependency on any other local repository source file. Looking at upstream, the only local include they have introduced is on `compat.h`, which is a helper include that has no other local dependencies. If we wanted to re-use `new_handler_terminate` we'd at a minimum need to include `util.h`, which then means depending on `libbitcoin_util.a` anywhere prevector is used. That *appears* to be the case currently though...

Reply:
I don't see any problem with moving ``new_handler_terminate`` into util.cpp.